ast/parser.go

```diff
@@ -556,13 +557,52 @@ func (p *Parser) parseRules() []*Rule {
 		return []*Rule{&rule}
 	}
 
-	if p.s.tok == tokens.LBrace {
+	hasIf := false
```

Comment:
Here we change how rule heads are parsed. We can make believe that the new tokens are ordinary tokens, since the future keyword import is what changes the parser behaviour. If the future import isn't there, these will never show up in the code processing.